lib/statistics/statistical_test/chi_squared_test.rb

```diff
@@ -37,6 +37,59 @@ def self.goodness_of_fit(alpha, expected, observed)
           alternative: p_value <= alpha,
           confidence_level: 1 - alpha }
       end
+
+      # The following three functions serve to calculate a test of independence for contingency
+      # tables of the type
+      #
+      #      A   B
+      # X   20  18
+      # Y    7  35
+      #
+      # They have been tested using 2x2 and 3x3 tables.
+      #
+      def self.test_of_independence(alpha, observed)
+        expected = calculate_expected(observed)
+        df = (observed.size - 1) * (observed.transpose.size - 1)
+        chi_score = calculate_chi_score(observed, expected)
```

Comment:
I think the chi score calculated with `chi_statistic` works as well, so no need for another method.

Reply:
This is a bit of a tougher one: `chi_statistic` currently works for single values or flat arrays, but I am using a nested array for the contingency table. I have tried to implement the contingency table as Matrix before, but this created other problems... It would be the cleaner solution, though, I guess? This way, the `chi_statistic` function could be made to work for all three types.

Reply:
gotcha, so prefer to keep this method private so it can be updated without breaking external usage.